Doc/Manual/Javascript.html

```diff
@@ -89,6 +89,8 @@ <H3><a name="Javascript_running_swig">28.2.1 Running SWIG</a></H3>
 <pre>
 $ swig -c++ -javascript -jsc example.i</pre>
 </div>
+<p>Unlike JavaScriptCore, V8 is a C++ API, and thus, the generated modules
+must be generated and compiled as C++.</p>
```

Comment:
I think this is potentially misleading, as "must be generated [,,,] as C++" seems to suggest you need to use `swig -c++` which is wrong.  Try the `simple` example:

```
$ cd Examples/javascript/simple
$ make --no-print-directory
make -f ../../Makefile SRCDIR='' CXXSRCS='' \
SRCS='example.c' SWIG_LIB_DIR='../../../Lib' SWIGEXE='../../../swig' \
SWIGOPT='-DV8_VERSION=0x031110' TARGET='example' INTERFACE='example.i' javascript_wrapper
env SWIG_LIB=../../../Lib  ../../../swig -javascript -DV8_VERSION=0x031110 -node -o example_wrap.cxx example.i
make -f ../../Makefile SRCDIR='' CXXSRCS='' \
SRCS='example.c' SWIG_LIB_DIR='../../../Lib' SWIGEXE='../../../swig' \
SWIGOPT='-DV8_VERSION=0x031110' TARGET='example' INTERFACE='example.i' javascript_build
sed -e 's|$srcdir|./|g' binding.gyp.in > binding.gyp
MAKEFLAGS= node-gyp --loglevel=silent configure build 1>>/dev/null
make -f ../../Makefile SRCDIR='' TARGET='example' javascript_run
env NODE_PATH=$PWD:  node runme.js 
GCD of x and y is=21
Global variable Foo=3
Variable Foo changed to=3.1415926
```

Note no `-c++` on the SWIG command line (SWIG did use to use `-c++` here when you started work on this, but I fixed that in d3b17437b12625bd31118f40e06a3bb28b6c5fda).

If "generated" is trying to say you need to call the result `.cxx` or similar, that's not accurate either as you could let SWIG generate `example_wrap.c` and use `-xc++` on the compiler command like (with GCC and clang at least).

The key point here is you need to **compile** the generated wrapper as C++ for V8, which we actually already say below.

Reply:
Fixed.

Reply:
Thanks.

Just FYI, I remembered b508598a42b80f9e63f6ebbaf04fdbfee30ea2a2 - before that you would need to run swig with `-c++` when generating wrappers for node (or the undocumented `-c++out` option, or hack around the problem like our testsuite did by running swig with `-v8 -DBUILDING_NODE_EXTENSION=1` instead of `-node`).  So your original change here probably was essentially correct when it was written.